client/src/mapView.js

```diff
@@ -10,15 +10,57 @@ var MapView = React.createClass({
     router: React.PropTypes.func
   },
 
-  render () {
+  // this is called after the first reder of the component
+  componentDidMount () {
     var {origin, destination} = this.context.router.getCurrentParams();
 
+    var start = this.getLatLong(origin);
+    var end = this.getLatLong(destination);
+
+    var map = this.initializeMap(start);
+    this.calcRoute(start, end, map);
+  },
+
+  // turns a lat/long string into a google maps LatLong Object
+  getLatLong (location) {
+    return new google.maps.LatLng(location.split(',')[0], location.split(',')[1]);
```

Comment:
Please add line number in future. If you are referring to getLatLong, it will return an object that the google maps API expects when creating a route